### ISSUE TITLE
[v1.0.0] Getting "VK error" in Kodi 17.6 on Linux Mint 19.1

### ISSUE THREAD

**Issue by SingleMalt2104** [tommistolercz/plugin.video.vk#66]
Hello Tomas Mistoler [tommistolercz],

I was looking forward to use your add-on "plugin.video.vk" for Kodi 17.6 on my Linux Mint 19.1 Xfce, which is based on Ubuntu 18.04, but sadly run into the below mentioned problems. I am not sure if my occuring "VK error" in Kodi 17.6 is an user error on my end or if something has changed in your add-on since its release state of November 2018. I hope you continuously have the time to fix/update your add-on if things change with either the VK-API or something else. Looking at the youtube-addon of bromix, there is always something to edit/fix, in order to keep the add-on working properly. By the way, has your add-on still beta status or has it already reached final release status?

I am getting an "VK Error" every time I try to open/use folders/functions like:

- Search [error occurs directly after executing search entry]
- My videos [error occurs directly after trying to open the "My videos" main folder]
- My video albums [error occurs directly after trying to open the album folder]
- My communities [error occurs directly after trying to open the folder of the community]
- My likes [error occurs directly after trying to open the folder "My likes"]

Bottom line: I am able to successfully login with my credentials from VK.com into "plugin.video.vk" but afterwards I can't use anything of your advertised features in the add-on itself. Could this be of missing access rights for the add-on, though it says in the VK.com settings that it has those access rights already. I mean it reads so in the "apps" description of the add-on in VK.com?

Oddly enough I can like/unlike communities in Kodi that I had earlier subscribed to in VK.com directly. So, where could I start fixing things on my end if this is the case? Sorry to bother you if my problem is truly just on my end.

Here is the kodi.log for your convenience:
[kodi.log](https://github.com/tommistolercz/plugin.video.vk/files/2718633/kodi.log)

Best regards and Happy new Year 2019, SingleMalt2104

**Comment by tommistolercz** [comment]
Hello SingleMalt2104, 

thanks a lot for your interest and the issue reports. I apologize for my late response.

The problems you're struggling with are 100% on my side (of course ;-) and are related to the incorrect handling of Unicode in addon's code (in Python this is quite a pain as I have found out).

As you said, you can normally run the add-on, authorize, display menu... everything works fine, unless there comes a string containing non-ascii characters and this one is used (means displayed, read, saved, passed to, etc...) without proper decoding/encoding. (just find in your Kodi.log file the lines containing `plugin.video.vk:` and you'll see those damned `UnicodeEncodeError`s.

In the coming days I will finish the new release (v1.2.0), which I hope will fix these issues.

**Comment by SingleMalt2104** [comment]
Hello tommistolercz,

yes I am very interested in your plugin, as many video bloggers on youtube nowadays (I am a German resident) are having severe censorship problems. So they try to use more open-minded (censor-free) social networks where free-speech is still granted. At least for now this seems to be the case with VK.com. I hope it stays this way. ;)

And yes: After using the reset function in the plugin I am able to enter my credentials and login successfully afterwards. After that I see the complete folder structure as shown in your picture samples. 

But from hereon out I cannot do much anymore. A search is conducted but shows no results which I could select. It shows only the amount of results. I can't even enter the folders "My Videos", "Video Albums", "My Likes". I can enter the folder "My Communities" but am getting instantly an "VK.com error" on choosing and entering any one of them.

So I am looking forward for your next release (v1.2.0) which will hopefully fix said problems. Thanks for your hard work by the way. I am betting there are others out there who would also gladly use your plugin but shy away to give you any feedback. That's not how it works. ;)

best regards, SingleMalt2104

**Comment by tommistolercz** [comment]
Hello SingleMalt2104,

FYI a new version (v1.2.0) is out, I hope it will work for you. ;-)
I'd appreciate your feedback. Thx.

**Comment by SingleMalt2104** [comment]
Hello Tomas Mistoler [tommistolercz],

I posted already here a comment to v1.2.0.0: https://github.com/tommistolercz/plugin.video.vk/issues/57
as "Fix /playvideo raising video resolving error due to unhandled cookies expiration (#57)" seems to be my main problem now, but I should avoid double-posts so lets discuss here. So I quote my post here:

"First of all thank you for releasing v1.2.0. of your plugin. Now I was finally able to open and view/play some of my videos. I even could play a few videos of my subscribed communities after entering their sub-folders.

Suddenly I am confronted with the error message "Resolving error". As you already seem to know why this occurs ("[...] due to unhandled cookies expiration" #57) I will simply wait for a follow up version of your plugin where said problem(s) will have been fixed. Thank you for your continuing efforts."

best regards, SingleMalt2104
